core/src/main/java/org/testcontainers/containers/output/Slf4jLogConsumer.java

```diff
@@ -18,21 +23,44 @@ public Slf4jLogConsumer withPrefix(String prefix) {
         return this;
     }
 
+    public Slf4jLogConsumer withMdc(String key, String value) {
+        mdc.put(key, value);
+        return this;
+    }
+
+    public Slf4jLogConsumer withMdc(Map<String, String> mdc) {
+        this.mdc.putAll(mdc);
+        return this;
+    }
+
     @Override
     public void accept(OutputFrame outputFrame) {
         OutputFrame.OutputType outputType = outputFrame.getType();
 
         String utf8String = outputFrame.getUtf8String();
         utf8String = utf8String.replaceAll(FrameConsumerResultCallback.LINE_BREAK_AT_END_REGEX, "");
-        switch (outputType) {
-            case END:
-                break;
-            case STDOUT:
-            case STDERR:
-                logger.info("{}{}: {}", prefix, outputType, utf8String);
-                break;
-            default:
-                throw new IllegalArgumentException("Unexpected outputType " + outputType);
+
+        Map<String, String> originalMdc = MDC.getCopyOfContextMap();
+        MDC.setContextMap(mdc);
+        try {
+            switch (outputType) {
+                case END:
+                    break;
+                case STDOUT:
+                    logger.info("{}{}", prefix.isEmpty() ? "" : (prefix + ": "), utf8String);
+                    break;
+                case STDERR:
+                    logger.error("{}{}", prefix.isEmpty() ? "" : (prefix + ": "), utf8String);
```

Comment:
This feels a tad risky; STDERR isn't solely used for 'error' output in practice, so it might cause confusion to map it to the logger's error level.

I think it's one of those situations where there's no right answer...

Reply:
i'm not sure there would be confusion. standard out and standard error are different output streams. IMO it's more confusing to have them on the same log level because they are different streams.

Reply:
That's the name, but the [specification](https://pubs.opengroup.org/onlinepubs/9699919799/functions/stderr.html) is for it to carry diagnostic output, and that's what it is used for in real life. I don't think we should by default treat it differently from stdout, which is why they are both currently mapped to the same log level.

How about this: we could have an opt-in parameter to control whether stderr is mapped to error level logs. This way, people can choose what's best for them according to how the specific container behaves.